doc/fpmsyncd/frr_sonic_protobuf_channel.md

```diff
@@ -0,0 +1,232 @@
+To enable the FPM Protobuf module in FRR, FRR needs to be compiled with the `--enable-dplane-pb` and `--enable-protobuf` options.
+
+A new patch file is introduced to compile FRR with the above options. Specifically, this patch adjusts the FRR build options specified within the FRR's *debian/rules* file as follows:
+
+* The option `--disable-protobuf` is replaced with `--enable-protobuf`
+* The option `--enable-dplane-pb` is added
+
+```diff
+...
+
+override_dh_auto_configure:
+	$(shell dpkg-buildflags --export=sh); \
+	dh_auto_configure -- \
+		--sbindir=/usr/lib/frr \
+		--with-vtysh-pager=/usr/bin/pager \
+		--libdir=/usr/lib/$(DEB_HOST_MULTIARCH)/frr \
+		--with-moduledir=/usr/lib/$(DEB_HOST_MULTIARCH)/frr/modules \
+		LIBTOOLFLAGS="-rpath /usr/lib/$(DEB_HOST_MULTIARCH)/frr" \
+		--disable-dependency-tracking \
+		\
+		$(CONF_RPKI) \
+		$(CONF_LUA) \
+		$(CONF_PIM6) \
+		--with-libpam \
+		--enable-doc \
+		--enable-doc-html \
+		--enable-snmp \
+		--enable-fpm \
+-		--disable-protobuf \
++		--enable-protobuf \
+		--disable-zeromq \
+		--enable-ospfapi \
+		--enable-bgp-vnc \
+		--enable-multipath=256 \
+		\
+		--enable-user=frr \
+		--enable-group=frr \
+		--enable-vty-group=frrvty \
+		--enable-configfile-mask=0640 \
+		--enable-logfile-mask=0640 \
++		--enable-dplane-pb \
+		# end
+
+...
+```
+
+#### 7.3.2. Load the FPM Protobuf module on zebra startup
+
+To enable the FPM Protobuf module in FRR, the FRR zebra daemon should be started with the `-M dplane_fpm_pb` option.
+The command line options used to start zebra are specified in the *supervisord.conf.j2* template file, located within the *dockers/docker-fpm-frr/frr/supervisord/* directory of the *sonic-buildimage* repository.
+
+We modify the template *supervisor.conf.j2* file to add the `-M dplane_fpm_pb` option:
+
+```jinja2
+[program:zebra]
+command=/usr/lib/frr/zebra -A 127.0.0.1 -s 90000000 -M dplane_fpm_nl -M snmp --asic-offload=notify_on_offload -M dplane_fpm_pb
+```
+
+Including the `-M dplane_fpm_pb` option ensures that zebra loads the FPM Protobuf module during startup.
+
+## 8. Testing Requirements/Design  
+
+### 8.1. Unit Test cases  
+
+* **Test case 1:** Receiving an SRv6 VPN route encoded using the Protobuf format
+  1. Generate an FPM message containing an SRv6 VPN route encoded using Protobuf.
+  2. Call `FpmLink::processFpmMessage()` with the generated FPM message.
+  3. Verify the delivery of the FPM message to RouteSync by checking if `RouteSync.onMsgRaw()` is invoked.
+  4. Ensure that the route is added to the ROUTE_TABLE of APPL_DB.
+
+* **Test case 2:** Receiving an SRv6 SID encoded using the Protobuf format
+  1. Generate an FPM message containing an SRv6 SID encoded using Protobuf.
+  2. Call `FpmLink::processFpmMessage()` with the generated FPM message.
+  3. Verify the delivery of the FPM message to RouteSync by checking if `RouteSync.onMsgRaw()` is invoked.
+  4. Ensure that the route is added to the SRV6_MY_SID_TABLE of APPL_DB.
```

Comment:
Any out of sequence issue when some information of MY_SID_TABLE uses NL and other information like SRV6 SID uses Protobuf?